### PR TITLE
Add TODO test cases

### PR DESCRIPTION
**Pull request by oxarbitrage**
Added 2 test cases.

### CODE REVIEW

File: src/test/coldreward_tests.cpp [hunk 1]
```diff
@@ -828,7 +828,124 @@ BOOST_AUTO_TEST_CASE(checkpoints_rollback)
...
+    BOOST_REQUIRE_EQUAL(ranges.size(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr).size(), 4);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[0].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[0].getEnd(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[1].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[1].getEnd(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[2].getStart(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[2].getEnd(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[3].getStart(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[3].getEnd(), 2);
```
Comment (oxarbitrage) [inline src/test/coldreward_tests.cpp]: @TheQuantumPhysicist can you take a look here when you can ? 

I cant understand why in this case there is a closing [2,2] while the balance is still above threshold.

Comment (TheQuantumPhysicist) [reply src/test/coldreward_tests.cpp:947]: I can't find this anymore in my version. You're probably looking at an older commit?

File: src/test/coldreward_tests.cpp [hunk 2]
```diff
@@ -967,7 +967,124 @@ BOOST_AUTO_TEST_CASE(checkpoints_rollback)
...
+    BOOST_REQUIRE_EQUAL(ranges.at(addr2).size(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr2)[0].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr2)[0].getEnd(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr2)[1].getStart(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr2)[1].getEnd(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr3).size(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr3)[0].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr3)[0].getEnd(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr3)[1].getStart(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr3)[1].getEnd(), 2);
```
Comment (oxarbitrage) [inline src/test/coldreward_tests.cpp]: I was not expecting this, in my understanding the end of the first range for address3 should be 2 and we should not have a second range [2,2] for it.

File: src/test/coldreward_tests.cpp [hunk 3]
```diff
@@ -967,7 +967,124 @@ BOOST_AUTO_TEST_CASE(checkpoints_rollback)
...
+
+    BOOST_CHECK_EQUAL(balances.at(addr), 20000 * COIN);
+    BOOST_REQUIRE_EQUAL(ranges.size(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr).size(), 3);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[0].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[0].getEnd(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[1].getStart(), 1);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[1].getEnd(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[2].getStart(), 2);
+    BOOST_REQUIRE_EQUAL(ranges.at(addr)[2].getEnd(), 2);
```
Comment (oxarbitrage) [inline src/test/coldreward_tests.cpp:1102]: Not expecting to see this [2,2] in this case.

File: src/test/coldreward_tests.cpp (hunk 2 above)
Comment (oxarbitrage) [reply src/test/coldreward_tests.cpp]: The multiplier changes, that is the reason of the extra range. Test now consider this at https://github.com/ghost-coin/ghost-core/pull/52/commits/668da0d103f42ce19bc58cc2b773ccf1db9e1fba

File: src/test/coldreward_tests.cpp (hunk 3 above)
Comment (oxarbitrage) [reply src/test/coldreward_tests.cpp:1102]: https://github.com/ghost-coin/ghost-core/pull/52/commits/668da0d103f42ce19bc58cc2b773ccf1db9e1fba

**Pull request merged**